apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/dot-palette-contentlets/dot-palette-contentlets.component.ts

```diff
@@ -0,0 +1,161 @@
+    styleUrls: ['./dot-palette-contentlets.component.scss']
+})
+export class DotPaletteContentletsComponent implements OnInit, OnChanges {
+    @ViewChild('filterInput', { static: true })
+    filterInput: DotPaletteInputFilterComponent;
+
+    @Input() contentTypeVariable: string;
+    @Input() languageId: string;
+    @Output() hide = new EventEmitter();
+    items: DotCMSContentlet[] | DotCMSContentType[] = [];
+    isFormContentType: boolean;
+    hideNoResults = true;
+
+    private _ESParams: queryEsParams;
+    private _filter: string;
+
+    constructor(
+        public paginatorESService: DotPaginatorESContentService,
+        public paginationService: PaginatorService,
+        private dotContentletEditorService: DotContentletEditorService
+    ) {}
+
+    ngOnInit(): void {
+        this._ESParams = {
+            itemsPerPage: 15,
+            lang: this.languageId || '1',
+            query: ''
+        };
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes?.contentTypeVariable?.currentValue) {
+            this.isFormContentType = changes?.contentTypeVariable?.currentValue === 'forms';
+
+            if (this.isFormContentType) {
+                this.paginationService.url = `v1/contenttype`;
+                this.paginationService.paginationPerPage = 15;
+                this.paginationService.sortField = 'modDate';
+                this.paginationService.setExtraParams('type', 'Form');
+                this.paginationService.sortOrder = 1;
+            }
+
+            this.loadData();
+        }
+    }
+
+    /**
+     * Loads data through pagination service
+     *
+     * @param LazyLoadEvent [event]
+     * @memberof DotPaletteContentletsComponent
+     */
+    loadData(event?: LazyLoadEvent): void {
+        if (this.isFormContentType) {
+            this.paginationService.setExtraParams(
+                'filter',
+                this.filterInput.searchInput.nativeElement.value
+            );
+            this.paginationService
+                .getWithOffset((event && event.first) || 0)
+                .pipe(take(1))
+                .subscribe((data: DotCMSContentType[] | DotCMSContentlet[]) => {
+                    data.forEach((item) => (item.contentType = item.variable = 'FORM'));
+                    this.items = data;
+                    this.hideNoResults = !!data?.length;
+                });
+        } else {
+            this.paginatorESService
+                .get({
+                    ...this._ESParams,
+                    lang: this.languageId || '1',
+                    filter: this._filter || '',
+                    offset: (event && event.first) || 0,
+                    query: `+contentType: ${this.contentTypeVariable}`
+                })
+                .pipe(take(1))
+                .subscribe((data: DotCMSContentType[] | DotCMSContentlet[]) => {
+                    this.items = data;
+                    this.hideNoResults = !!data?.length;
```

Comment:
Can you just use in the UI `!!items?.lenght`? instead of handling another property for the state.

Reply:
I'll ignore this, since it will be removed once the state management store gets implemented